inc/functions.php

```diff
@@ -99,7 +99,32 @@ function srm_max_redirects_reached() {
  * @return array
  */
 function srm_get_valid_status_codes() {
-	return apply_filters( 'srm_valid_status_codes', array( 301, 302, 303, 307, 403, 404 ) );
+	return apply_filters( 'srm_valid_status_codes', array_keys( srm_get_valid_status_codes_data() ) );
+}
+
+/**
+ * Get valid HTTP status codes and their labels.
+ *
+ * @since  2.0.0
+ * @return array
+ */
+function srm_get_valid_status_codes_data() {
+	$status_codes = array(
+		301 => esc_html__( 'Moved Permanently', 'safe-redirect-manager' ),
+		302 => esc_html__( 'Found', 'safe-redirect-manager' ),
+		303 => esc_html__( 'See Other', 'safe-redirect-manager' ),
+		307 => esc_html__( 'Temporary Redirect', 'safe-redirect-manager' ),
+		403 => esc_html__( 'Forbidden', 'safe-redirect-manager' ),
+		404 => esc_html__( 'Not Found', 'safe-redirect-manager' ),
+		410 => esc_html__( 'Gone', 'safe-redirect-manager' ),
+	);
+
+	$additional_status_codes = apply_filters(
+		'srm_additional_status_codes',
+		array()
+	);
+
+	return $status_codes + array_diff_key( $additional_status_codes, $status_codes );
```

Comment:
I *think* this is okay but I just want to double check that PHP 8 won't get mad about `+` as an array operator.

Reply:
I tested `+` against `php-8.0.0rc4` and I don't see any problem. The site I used to test is: https://3v4l.org/